src/porepy/fracs/msh_2_grid.py

```diff
@@ -467,3 +492,93 @@ def create_embedded_line_grid(
     # Add mapping to global point numbers.
     g.global_point_ind = glob_id[sort_ind]
     return g
+
+
+def tag_grid(
+    sd: pp.Grid,
+    phys_names: dict[int, str],
+    cell_info: dict[str, np.ndarray],
+) -> pp.Grid:
```

Comment:
I suspect this could be refined slightly by using a typevar or whatever saying that the return is the same subclass of Grid as sd is. Might not be worth it, though.